SharePointFramework/ProjectWebParts/src/components/ProjectPhases/changeWelcomePage.ts

```diff
@@ -1,20 +1,27 @@
 import initSpfxJsom, { ExecuteJsomQuery } from 'spfx-jsom'
+import { IPhaseSitePageModel } from './types'
 
 /**
  * Change phase
  *
  * @param {string} phaseName
  * @param {string} absoluteUrl absoluteurl
+ * @param {IPhaseSitePageModel[]} phaseSitePages Phase SitePages
  */
 export const changeWelcomePage = async (
   phaseName: string,
-  absoluteUrl: string
+  absoluteUrl: string,
+  phaseSitePages?: IPhaseSitePageModel[],
 ) => {
   try {
-    const spfxJsomContext = await initSpfxJsom(absoluteUrl)
-    spfxJsomContext.jsomContext.web['get_rootFolder']()['set_welcomePage'](`SitePages/${phaseName}.aspx`)
-    spfxJsomContext.jsomContext.web.update()
-    await ExecuteJsomQuery(spfxJsomContext.jsomContext)
+    const phaseSitePage = phaseSitePages.filter((phaseSitePage) => phaseSitePage.title === phaseName)[0]
```

Comment:
Perhaps use `find` from `underscore`? Or use [phaseSitePage]. Consistency is key.

https://underscorejs.org/#find